py-polars/polars/lazyframe/frame.py

```diff
@@ -1994,6 +1995,88 @@ def cache(self) -> Self:
         """Cache the result once the execution of the physical plan hits this node."""
         return self._from_pyldf(self._ldf.cache())
 
+    def cast(
+        self,
+        dtypes: Mapping[ColumnNameOrSelector, PolarsDataType] | PolarsDataType,
+        *,
+        strict: bool = True,
+    ) -> Self:
+        """
+        Cast LazyFrame column(s) to the specified dtype(s).
+
+        Parameters
+        ----------
+        dtypes
+            Mapping of column names (or selector) to dtypes, or a single dtype
+            to which all columns will be cast.
+        strict
+            Throw an error if a cast could not be done (for instance, due to an
+            overflow).
+
+        Examples
+        --------
+        >>> from datetime import date
+        >>> lf = pl.LazyFrame(
+        ...     {
+        ...         "foo": [1, 2, 3],
+        ...         "bar": [6.0, 7.0, 8.0],
+        ...         "ham": [date(2020, 1, 2), date(2021, 3, 4), date(2022, 5, 6)],
+        ...     }
+        ... )
+
+        Cast specific frame columns to the specified dtypes:
+
+        >>> lf.cast({"foo": pl.Float32, "bar": pl.UInt8}).collect()
+        shape: (3, 3)
+        ┌─────┬─────┬────────────┐
+        │ foo ┆ bar ┆ ham        │
+        │ --- ┆ --- ┆ ---        │
+        │ f32 ┆ u8  ┆ date       │
+        ╞═════╪═════╪════════════╡
+        │ 1.0 ┆ 6   ┆ 2020-01-02 │
+        │ 2.0 ┆ 7   ┆ 2021-03-04 │
+        │ 3.0 ┆ 8   ┆ 2022-05-06 │
+        └─────┴─────┴────────────┘
+
+        Cast all frame columns to the specified dtype:
+
+        >>> lf.cast(pl.Utf8).collect().to_dict(False)
+        {'foo': ['1', '2', '3'],
+         'bar': ['6.0', '7.0', '8.0'],
+         'ham': ['2020-01-02', '2021-03-04', '2022-05-06']}
+
+        Use selectors to define the columns being cast:
+
+        >>> import polars.selectors as cs
+        >>> lf.cast({cs.numeric(): pl.UInt32, cs.temporal(): pl.Utf8}).collect()
+        shape: (3, 3)
+        ┌─────┬─────┬────────────┐
+        │ foo ┆ bar ┆ ham        │
+        │ --- ┆ --- ┆ ---        │
+        │ u32 ┆ u32 ┆ str        │
+        ╞═════╪═════╪════════════╡
+        │ 1   ┆ 6   ┆ 2020-01-02 │
+        │ 2   ┆ 7   ┆ 2021-03-04 │
+        │ 3   ┆ 8   ┆ 2022-05-06 │
+        └─────┴─────┴────────────┘
+
+        """
+        if not isinstance(dtypes, Mapping):
+            return self.select(F.all().cast(dtypes, strict=strict))
```

Comment:
I can imagine we want this functionality available on the Rust side as well, though obviously it would need to be a different method. `cast_single_dtype`?

Not sure how far the parity between Rust and Python needs to go (I'd classify this one as a gray area).

Reply:
> obviously it would need to be a different method. cast_single_dtype?

Or `cast_all`? But yes, I wasn't convinced on the Rust side if we want to eat the extra API surface given the relative ease of `select( all().cast(Utf8) )`.

Reply:
I like `cast_all`!

Basically, if the functionality is worth having on the Python side, it's worth having on the Rust side. Mostly, anyway.

In this case, I think it's worth having.

Reply:
"Sold" - I'll throw it in :)

Reply:
Done! ✌️  (Also just rebased onto the latest code to pick up the `wasm` fix for CI)